### PR TITLE
buildkite: add permissions to observablt-robots-automation

### PR DESCRIPTION
**Pull request by v1v**
## What does this PR do?

Use a machine account which it's member of https://github.com/orgs/elastic/teams/observablt-robots-automation

## Checklist

- [ ] This is an enhancement of existing features, or a new feature in existing plugins
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] I have made corresponding changes to the documentation
- [ ] This is a bugfix
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] I have added tests that would fail without this fix
- [ ] This is a new plugin
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)
  - [ ] My code follows the [style guidelines of this project](https://github.com/elastic/apm-agent-java/blob/main/CONTRIBUTING.md#java-language-formatting-guidelines)
  - [ ] I have made corresponding changes to the documentation
  - [ ] I have added tests that prove my fix is effective or that my feature works
  - [ ] New and existing [**unit** tests](https://github.com/elastic/apm-agent-java/blob/main/CONTRIBUTING.md#testing) pass locally with my changes
  - [ ] I have updated [supported-technologies.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/docs/supported-technologies.asciidoc)
  - [ ] Added an API method or config option? Document in which version this will be introduced
  - [ ] Added an instrumentation plugin? Describe how you made sure that old, non-supported versions are not instrumented by accident.
- [x] This is something else
  - [ ] I have updated [CHANGELOG.asciidoc](https://github.com/elastic/apm-agent-java/blob/main/CHANGELOG.asciidoc)

### CODE REVIEW

File: catalog-info.yaml [hunk 1]
```diff
@@ -27,6 +27,8 @@ spec:
           access_level: MANAGE_BUILD_AND_READ
         observablt-robots:
           access_level: MANAGE_BUILD_AND_READ
+        observablt-robots-automation:
```
Comment (reakaleek) [inline catalog-info.yaml:30]: In a previous PR you used `BUILD_AND_READ`. Would that be also sufficient here?

**Review by reakaleek — APPROVED** [review]

File: catalog-info.yaml [hunk 2]
```diff
@@ -27,6 +27,8 @@ spec:
...
+          access_level: MANAGE_BUILD_AND_READ
```
Comment (v1v) [inline catalog-info.yaml:31]:
```suggestion
          access_level: BUILD_AND_READ
```

**Comment by SylvainJuge** [comment]
Is there any reason to not always have `BUILD_AND_READ` for `observablt-robots-automation` ?

From what I understand `observablt-robots-automation` is only used for automation and machine jobs, and I don't think we need to allow managing the jobs themselves, which I would have expected to be a human-only responsibility.

File: catalog-info.yaml [hunk 3]
```diff
@@ -59,6 +61,8 @@ spec:
...
```
Comment (v1v) [inline catalog-info.yaml:65]:
```suggestion
          access_level: BUILD_AND_READ
```

File: catalog-info.yaml [hunk 4]
```diff
@@ -91,6 +95,8 @@ spec:
...
```
Comment (v1v) [inline catalog-info.yaml:99]:
```suggestion
          access_level: BUILD_AND_READ
```

File: catalog-info.yaml [hunk 5]
```diff
@@ -124,6 +130,8 @@ spec:
...
```
Comment (v1v) [inline catalog-info.yaml:134]:
```suggestion
          access_level: BUILD_AND_READ
```

**Comment by v1v** [comment]
`BUILD_AND_READ` is now the default behaviour

**Pull request merged**